src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -1836,9 +1832,7 @@ class StubGenerator: public StubCodeGenerator {
     __ bne(t1, scratch_src_klass, L_failed);
 
     // if src->is_Array() isn't null then return -1
-    // i.e. (lh >= 0)
-    __ test_bit(t0, lh, 31);
-    __ beqz(t0, L_failed);
+    __ bgez(t0, L_failed);
```

Comment:
Should this be: `__ bgez(lh, L_failed);` ?

Reply:
Fixed.